container_pipeline/workers/test.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python
+
+import hashlib
+import json
+import logging
+import os
+
+from container_pipeline.utils import Build, get_job_name
+from container_pipeline.lib.log import load_logger
+from container_pipeline.lib.openshift import Openshift, OpenshiftError
+from container_pipeline.workers.base import BaseWorker
+
+
+class TestWorker(BaseWorker):
+    """Test Worker"""
+
+    def __init__(self, logger=None, sub=None, pub=None):
+        super(TestWorker, self).__init__(logger=logger, sub=sub, pub=pub)
+        self.openshift = Openshift(logger=self.logger)
+
+    def test(self, job):
+        """Runs openshift test job"""
+        namespace = get_job_name(job)
+        project = hashlib.sha224(namespace).hexdigest()
+
+        try:
+            self.openshift.login()
+            build_id = self.openshift.build(project, 'test')
+            if not build_id:
+                return False
+        except OpenshiftError as e:
+            logger.error(e)
+            return False
+
+    def handle_test_success(self, job):
+        """Handle build success for job"""
+        self.logger.debug("Test is successful going for next job")
+
+    def handle_test_failure(self, job):
+        """Handle build failure for job"""
+        self.queue.put(json.dumps(job))
+        self.logger.info(
+            "Test is not successful putting it to failed build tube")
+        appid = job['appid']
+        jobid = job['jobid']
+        desired_tag = job['desired_tag']
+        project = appid + "/" + jobid + ":" + desired_tag
+        test_logs_file = os.path.join(job['logs_dir'], 'test_logs.txt')
+        self.notify_build_failure(
+            get_job_name(job), job['notify_email'], test_logs_file,
+            project, job['jobid'], job['TEST_TAG'])
```

Comment:
(in regard to above comment)

since you are not re-using the local variable, but accessing via `job` dict